mlflow/system_metrics/system_metrics_monitor.py

```diff
@@ -0,0 +1,128 @@
+"""Class for monitoring system stats."""
+
+import logging
+import os
+import threading
+
+from mlflow.system_metrics.metrics.cpu_monitor import CPUMonitor
+from mlflow.system_metrics.metrics.gpu_monitor import GPUMonitor
+
+_logger = logging.getLogger(__name__)
+
+
+class SystemMetricsMonitor:
```

Comment:
Should we define this as a context handler that can terminate its processing thread on __exit__ and initialize on __enter__? 
I'm curious about the chances of a zombie process here without a cleanup process (essentially, if ``finish()`` is not able to be called, this monitor thread will continue until the REPL is exited, right?

Reply:
Yea, exiting is an issue. It depends on the user workflow we support, context handler means we want to have things like:
```
with SystemMetricsMonitor() as monitor:
    monitor.start()
```
In the discussion, mostly we are thinking about a user journey that system metrics logging is automatically started at `mlflow.start_run()`, and not expecting users to call this class directly. If users want to adapt this to their custom workflow, they should control where `finish` is called.